erpnext/erpnext_integrations/taxjar_integration.py

```diff
@@ -0,0 +1,282 @@
+		'to_zip': shipping_address.pincode,
+		'to_city': shipping_address.city,
+		'to_street': shipping_address.address_line1,
+		'to_state': to_shipping_state,
+		'shipping': shipping,
+		'amount': doc.net_total,
+		'line_items': [item_dict]
+	}
+
+
+
+	return tax_dict
+
+def sanitize_error_response(response):
+	response = response.full_response.get("detail")
+	response = response.replace("_", " ")
+
+	sanitized_responses = {
+		"to zip": "Zipcode",
+		"to city": "City",
+		"to state": "State",
+		"to country": "Country"
+	}
+
+	for k, v in sanitized_responses.items():
+		response = response.replace(k, v)
+
+	return response
+
+
+def set_sales_tax(doc, method):
+	if not doc.items:
+		return
+
+	# Allow skipping calculation of tax for dev environment
+	# if taxjar_calculate_tax isn't defined in site_config we assume
+	# we DO want to calculate tax all the time.
+	if not TAXJAR_CALCULATE_TAX:
+		return
+
+	if doc.exempt_from_sales_tax or frappe.db.get_value("Customer", doc.customer, "exempt_from_sales_tax"):
+		for tax in doc.taxes:
+			if tax.account_head == TAX_ACCOUNT_HEAD:
+				tax.tax_amount = 0
+				break
+
+		doc.run_method("calculate_taxes_and_totals")
+		return
+
+	tax_dict = get_tax_data(doc)
+
+	if not tax_dict:
+		# Remove existing tax rows if address is changed from a taxable state/country
+		setattr(doc, "taxes", [tax for tax in doc.taxes if tax.account_head != TAX_ACCOUNT_HEAD])
+		return
+
+	tax_data = validate_tax_request(tax_dict)
+
+	if tax_data is not None:
+		if not tax_data.amount_to_collect:
+			setattr(doc, "taxes", [tax for tax in doc.taxes if tax.account_head != TAX_ACCOUNT_HEAD])
+		elif tax_data.amount_to_collect > 0:
+			# Loop through tax rows for existing Sales Tax entry
+			# If none are found, add a row with the tax amount
+			for tax in doc.taxes:
+				if tax.account_head == TAX_ACCOUNT_HEAD:
+					tax.tax_amount = tax_data.amount_to_collect
+
+					doc.run_method("calculate_taxes_and_totals")
+					break
+			else:
+				doc.append("taxes", {
+					"charge_type": "Actual",
+					"description": "Sales Tax",
+					"account_head": TAX_ACCOUNT_HEAD,
+					"tax_amount": tax_data.amount_to_collect
+				})
+
+			doc.run_method("calculate_taxes_and_totals")
+
+
+def validate_address(doc, address):
+	# Validate address using PyCountry
+	tax_dict = get_tax_data(doc)
+
+	if tax_dict:
+		# Validate address using TaxJar
+		validate_tax_request(tax_dict)
+
+
+def validate_tax_request(tax_dict):
+	"""Return the sales tax that should be collected for a given order."""
+
+	client = get_client()
+
+	try:
+		tax_data = client.tax_for_order(tax_dict)
+	except taxjar.exceptions.TaxJarResponseError as err:
+		frappe.throw(_(sanitize_error_response(err)))
+	else:
+		return tax_data
+
+
+def validate_state(address):
+	country_code = frappe.db.get_value("Country", address.get("country"), "code")
+
+	error_message = _("""{} is not a valid state! Check for typos or enter the ISO code for your state.""".format(address.get("state")))
```

Comment:
The message translation is incorrect. Please refer
https://frappe.io/docs/user/en/translations